### DISCUSSION TITLE
Python: Unexpected flow from RemoteFlowSource

### DISCUSSION THREAD

**Discussion by jorgectf** [github/codeql#5589] [Q&A] [answered]
I'm working with a fairly easy flow, where the `source` is `RemoteFlowSource` and the sinks are both `unsafe_dn` and `unsafe_filter`.

```py
@app.route("/normal")
def normal():
    unsafe_dn = "dc=%s" % request.args['dc']
    unsafe_filter = "(user=%s)" % request.args['username']

    srv = ldap3.Server('ldap://127.0.0.1', port=1337)
    conn = ldap3.Connection(srv, user=unsafe_dn, auto_bind=True)
    conn.search(unsafe_dn, unsafe_filter, attributes=[
                "testAttr1", "testAttr2"])
```

The thing is that, I get three different flows. The first one is the intended one (`request`(l.3) -> `request.args`(l3) -> `request.args['dc']`(l.3) -> `unsafe_dn`), but the other ones are related to `unsafe_filter` and make some strange jumps (`request`(l.9) directly jumps to `request.args`(l.10) & `request`(l.9) ->`request`(l.10)). 

![image](https://user-images.githubusercontent.com/46056498/113349610-9e680180-9338-11eb-967e-955dcab99e6b.png)

I guess [this comment](https://github.com/github/codeql/pull/5443#discussion_r605207553) is also related to this issue.
Complete unit tests and query: https://github.com/github/codeql/pull/5443

Thanks in advance for your time! :D

**Comment by yoff** [comment, marked as answer]
I think that `request` (l.9) -> `request` (l.10) (in the third path explanation) is use-use flow. That is, the reference to `request` on line 10 is "the next use" of that variable after the reference on line 9. Use-use flow is normally compressed out of the path explanations, but it may be shown here because we force is to show all subscript reads. However, it seems to also give you the version where it is compressed away in the second path explanation.

**Reply by jorgectf** [reply]
Thanks for the reply @yoff! How could that flow not be shown in the result? It's a bit weird to have two flows for the same sink and source.

**Reply by yoff** [reply]
I am afraid you can not do much about this immediately. There are some kinks in how we use the data flow library that we still need to iron out; I believe you are seeing consequences of this...

**Reply by jorgectf** [reply]
No problem then, I was afraid that the problem was on my side (and so the results would be weird), but since it belongs to the internal stuff, they will improve once the libs are polished up. Thanks @yoff for your time! Appreciate it :)

**Reply by jorgectf** [reply]
By the way, would you like me to open an issue regarding this problem or it's something you are already looking into? 👍

**Reply by yoff** [reply]
If I am right about the cause, then it is definitely on the radar already :-)
